src/utils/user-agent.ts

```diff
@@ -19,5 +20,28 @@ export const buildUserAgent = () => {
     userAgentParts.push(`node ${process.version}`);
   }
 
+  if (config.sourceTag) {
+    userAgentParts.push(`source_tag=${normalizeSourceTag(config.sourceTag)}`);
+  }
+
   return userAgentParts.join('; ');
 };
+
+const normalizeSourceTag = (sourceTag: string) => {
+  if (!sourceTag) {
+    return;
+  }
+
+  /**
+   * normalize sourceTag
+   * 1. Lowercase
+   * 2. Limit charset to [a-z0-9_ ]
+   * 3. Trim left/right spaces
+   * 4. Condense multiple spaces to one, and replace with underscore
+   */
+  return sourceTag
+    .toLowerCase()
+    .replace(/[^a-z0-9_ ]/g, '')
+    .trim()
+    .replace(/[ ]+/g, '_');
```

Comment:
Minor nit -- I I think the regex is overkill here vs the more readable `.replaceAll(' ', '_')`

Reply:
I think we would need to update the library we're targeting with TypeScript in order to use `replaceAll()`. Also, would that not replace multiple spaces with multiple underscores? 

```javascript
const testString = "   test_string   lol   hehehe"
testString.replaceAll(' ', '_')
// '___test_string___lol___hehehe'
```

Reply:
My bad, you're right